### PR TITLE
frigate: 0.12.1 -> 0.13.1 

### PR DESCRIPTION
**Pull request by fleaz**
## Description of changes

Diff: https://github.com/blakeblackshear/frigate/compare/v0.12.1...v0.13.1

With a lot of help from @mweinelt :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fleaz** [comment]
I just deployed this on my local NAS and upgraded from 0.12.1 to 0.13.1
App appears to be working without problems, database migration went through without problems and all all recordings/clips are still there. But the log was getting flooded with 
```
Feb 04 13:24:03 bart python3.11[2314048]: [2024-02-04 13:24:03] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
Feb 04 13:24:03 bart python3.11[2314048]: [2024-02-04 13:24:03] frigate.record.maintainer      ERROR   : not enough values to unpack (expected 2, got 1)
```

I found this issue and after clearing /var/cache/frigate together with a restart, the errors are gone. So if they expect the folder to be empty after every restart (and therefore also after an upgrade) we maybe should also clear this on start? Not sure if systemd has some fancy features for this to get a clean folder on every start or if I should just add a "rm -rf /var/cache/frigate/*" as a `PreStart=` into the unit?

**Comment by fleaz** [comment]
Another small issue: The new "Export" tab is not working and throwing a weird JS stacktrace. Not really helpful and I also found nothing in the Frigate issue tracker.
 
2024-02-04T14:57:42,134975207+01:00

**Comment by risicle** [comment]
When I run the nixos test:

```
machine # [   25.828757] nginx-pre-start[1091]: nginx: [emerg] "proxy_cache" zone "frigate:api_cache" is unknown in /nix/store/4djm1j9vrwkf81dmv82yci62g0phm68j-nginx.conf:247
machine # [   25.830043] nginx-pre-start[1091]: nginx: configuration file /nix/store/4djm1j9vrwkf81dmv82yci62g0phm68j-nginx.conf test failed
machine # [   25.836714] systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
machine # [   25.837540] systemd[1]: nginx.service: Failed with result 'exit-code'.
machine # [   25.840213] systemd[1]: Failed to start Nginx Web Server.
machine # [   36.011188] systemd[1]: nginx.service: Scheduled restart job, restart counter is at 3.
machine # [   36.017594] systemd[1]: Starting Nginx Web Server...
machine # [   36.118974] nginx-pre-start[1139]: nginx: [emerg] "proxy_cache" zone "frigate:api_cache" is unknown in /nix/store/4djm1j9vrwkf81dmv82yci62g0phm68j-nginx.conf:247
machine # [   36.120486] nginx-pre-start[1139]: nginx: configuration file /nix/store/4djm1j9vrwkf81dmv82yci62g0phm68j-nginx.conf test failed
machine # [   36.130597] systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
machine # [   36.131643] systemd[1]: nginx.service: Failed with result 'exit-code'.
machine # [   36.136118] systemd[1]: Failed to start Nginx Web Server.
```

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8431,6 +8433,8 @@ self: super: with self; {
 
   node-semver = callPackage ../development/python-modules/node-semver { };
 
+  norfair = callPackage ../development/python-modules/norfair { };
+
   noise = callPackage ../development/python-modules/noise { };
 
   noiseprotocol = callPackage ../development/python-modules/noiseprotocol { };
```
Comment (mweinelt) [inline pkgs/top-level/python-packages.nix:8440]:
```suggestion
  noise = callPackage ../development/python-modules/noise { };

  noiseprotocol = callPackage ../development/python-modules/noiseprotocol { };

  norfair = callPackage ../development/python-modules/norfair { };
```

File: nixos/modules/services/video/frigate.nix [hunk 2]
```diff
@@ -125,39 +125,71 @@ in
...
         };
       };
+      proxyCachePath."frigate" = {
+        keysZoneSize = "10m";
+        keysZoneName = "api_cache";
+        maxSize = "10m";
+        inactive = "1m";
+        levels = "1:2";
+      };
       # Based on https://github.com/blakeblackshear/frigate/blob/v0.12.0/docker/rootfs/usr/local/nginx/conf/nginx.conf
```
Comment (mweinelt) [inline nixos/modules/services/video/frigate.nix:158]: Outdated reference

File: nixos/modules/services/video/frigate.nix [hunk 3]
```diff
@@ -125,39 +125,71 @@ in
...
         frigate-jsmpeg.servers = {
-          "127.0.0.1:8082" = {};
+          "127.0.0.1:8082" = { };
         };
         frigate-go2rtc.servers = {
-          "127.0.0.1:1984" = {};
+          "127.0.0.1:1984" = { };
         };
       };
+      proxyCachePath."frigate" = {
```
Comment (mweinelt) [inline nixos/modules/services/video/frigate.nix:151]:
```suggestion
      proxyCachePath."frigate" = {
        enable = true;
```

But I think it should be default enabled in the nginx module. Feels weird to have to enable subsections of a config.

**Comment by risicle** [comment]
All packages (but frigate itself) build happily on macos 12 x86_64.

File: nixos/tests/frigate.nix [hunk 4]
```diff
@@ -55,6 +56,6 @@ import ./make-test-python.nix ({ pkgs, lib, ...} :
 
     machine.succeed("curl http://localhost:5001")
```
Comment (mweinelt) [inline nixos/tests/frigate.nix:57]:
```suggestion
    machine.wait_for_open_port(80)

    machine.succeed("curl --fail http://localhost")
```

**Review by risicle — APPROVED** [review]

**Review by mweinelt — APPROVED** [review]

**Pull request merged**